packages/interactivity-router/src/index.js

```diff
@@ -102,12 +135,47 @@ window.addEventListener( 'popstate', async () => {
 	}
 } );
 
-// Cache the initial page using the intially parsed vDOM.
+// Initialize the router and cache the initial page using the initial vDOM.
+// Once this code is tested and more mature, the head should be updated for region based navigation as well.
+if ( process.env.IS_GUTENBERG_PLUGIN ) {
+	if ( navigationMode === 'fullPage' ) {
+		// Cache the scripts. Has to be called before fetching the assets.
+		[].map.call( document.querySelectorAll( 'script[src]' ), ( script ) => {
+			headElements.set( script.getAttribute( 'src' ), {
+				tag: script,
+				text: script.textContent,
+			} );
+		} );
+		await fetchHeadAssets( document, headElements );
+	}
+}
 pages.set(
 	getPagePath( window.location ),
 	Promise.resolve( regionsToVdom( document, { vdom: initialVdom } ) )
 );
 
+// Check if the link is valid for client-side navigation.
+const isValidLink = ( ref ) =>
+	ref &&
+	ref instanceof window.HTMLAnchorElement &&
+	ref.href &&
+	( ! ref.target || ref.target === '_self' ) &&
+	ref.origin === window.location.origin &&
+	! ref.pathname.startsWith( '/wp-admin' ) &&
+	! ref.pathname.startsWith( '/wp-login.php' ) &&
```

Comment:
Sometimes WordPress sites install core in a subdirectory, like `/wp/`. In such cases `home_url()` and `site_url()` are different, e.g. `http://example.com/` vs `https://example.com/wp/`. 

Additionally, in multisite subdirectory installs, there will also be path segment(s) before `/wp-admin`. 

Easiest way to handle that right now is to just check if the path includes the strings:

```suggestion
	! ref.pathname.includes( '/wp-admin' ) &&
	! ref.pathname.includes( '/wp-login.php' ) &&
```

This is not as robust as possible. But better to err on the side of not doing client-side navigation than doing it with an invalid target.

Reply:
I’m simply reading the discussion, but seeing this solution would not work for my sites, because for security I always change the default wp-admin url. Maybe check for a class name or other attribute in the dom instead?

Reply:
Actually, these paths should be passed from PHP so they don't have to be hard coded in JS.

Reply:
You're right. Maybe we can use [wp_interactivity_config](https://developer.wordpress.org/reference/functions/wp_interactivity_config/) to create a new `isAdmin` property that is passed to the client reading [is_admin](https://developer.wordpress.org/reference/functions/is_admin/) in PHP.

cc: @DAreRodz Worth adding this to the tracking issue as a follow-up task?

Reply:
I don't think it would be `is_admin()` since this would always just be `false` in the frontend context. I think we'd need to instead get an array of the special paths, including `get_admin_url()`, `wp_login_url()`, etc.

Reply:
I've added this as a task on the tracking issue.